### DISCUSSION TITLE
How to mix RTL and LTR characters in the same string when  using collection view source.

### DISCUSSION THREAD

**Discussion by danies8** [dotnet/maui#15517] [Q&A]
### Description

Hi,
If you can  offer some suggestion on how to mix RTL and LTR characters in the same string when  using
collection view source.

### Steps to Reproduce

1.Create Maui app
2.Add collection view to the main page as follow.
3.
`<CollectionView ItemsSource="{Binding FilesInfo}">
    <CollectionView.ItemTemplate>
        <DataTemplate>
            <Label Text="{Binding FileName}" />
        </DataTemplate>
    </CollectionView.ItemTemplate>
</CollectionView>`
4.Add data source to collection view :
file name 1: שלום.txt
file name 2:hello.txt
5. results:
file name 1: **txt.שלום**
file name 2:hello.txt

### Link to public reproduction project repository

no

### Version with bug

7.0.49

### Last version that worked well

6.0

### Affected platforms

Windows

### Affected platform versions

.

### Did you find any workaround?

no

### Relevant log output

```shell
no
```

**Comment by Xyncgas** [comment]
I heard it's encoding problem that people dealt with in C++ too
https://stackoverflow.com/questions/65736177/weird-behaviour-on-joining-mixed-right-to-left-and-left-to-right-language-string
TLDR, add the padding character between each transition to reset the direction of text for the unicode algorithm
